AG-8777 Use theme default axis grid color

olegat opened the pull request:
https://ag-grid.atlassian.net/browse/AG-8777

olegat committed:
AG-8777 Use theme default axis grid color

olegat committed:
AG-8777 Fix unused import build error

olegat committed:
AG-8777 Update snapshot

I've checked that this is intended output:
https://plnkr.co/edit/jz8nBE8jMaUg2W6V?open=main.js

The bottom axis options includes: `style: [ { lineDash: [0]} ]`

olegat committed:
AG-8777 Add minimal repo test case

alantreadway commented on packages/ag-charts-community/src/chart/mapping/prepare.ts:
```diff
@@ -350,6 +354,19 @@ function prepareAxis<T extends AxesOptionsTypes>(
+                }
+                return value;
+            });
+        }
```
Would it be possible to take a similar approach to `crossLines` above (i.e. JSON merge of array elements with theme options), rather than treating `gridLine.style` as an exception that is passed in as a parameter?

olegat replied:
Done. There are some nuanced differences between `crossLines` and `gridLine`, which I've explained in code comments.

olegat committed:
AG-8777 Code review feedback

Apply the gridLine.style theme in the same way as the crossLine theme.

olegat committed:
AG-8777 Fix failing unit tests

An empty grid line style `{}` mean "draw nothing", so ignore the theme
defaults in this case. Also add a test case specifically for this.

alantreadway commented on packages/ag-charts-community/src/chart/mapping/prepare.ts:
```diff
@@ -345,9 +347,27 @@ function prepareAxis<T extends AxesOptionsTypes>(
+            }
+            // Themes will normally only have one element in gridLineStyle[], but cycle through the array
+            // with `mod` anyway to make sure that we honour the theme's grid line style sequence.
+            const themeStyle: typeof userStyle = gridLineStyle[mod(index, gridLineStyle.length)];
```
Nit: What is `mod()`? It's more conventional to use the `%` operator in JS?

alantreadway replied:
(Just looked, `mod()` is overkill in this scenario where we are guaranteed that the values are positive integers)

olegat replied:
Done.

olegat committed:
AG-8777 Use % instead of mod

alantreadway reviewed (approved):
LGTM
